nibabel/nifti1.py

```diff
@@ -379,13 +381,107 @@ def write_to(self, fileobj, byteswap):
         fileobj.write(b'\x00' * (extstart + rawsize - fileobj.tell()))
 
 
+class Nifti1DicomExtension(Nifti1Extension):
+    """NIfTI1 DICOM header extension
+
+    This class is a thin wrapper around pydicom to read a binary DICOM
+    byte string. If pydicom is not available, it silently falls back to the
+    standard NiftiExtension class.
+    """
+    def __init__(self, code, content, parent_hdr=None):
+        """
+        Parameters
+        ----------
+        code : int|str
+          Canonical extension code as defined in the NIfTI standard, given
+          either as integer or corresponding label
+          (see :data:`~nibabel.nifti1.extension_codes`)
+        content : bytes|pydicom Dataset|None
+          Extension content - either a bytestring as read from the NIfTI file
+          header or an existing pydicom Dataset. If a bystestring, the content
+          is converted into a Dataset on initialization. If None, a new empty
+          Dataset is created.
+        parent_hdr : :data:`~nibabel.nifti1.Nifti1Header`
```

Comment:
Also, can you help with the ReStructuredText for linking to the pydicom Dataset class? Doesn't look like you're using the `conf.py` intersphinx mapping.